pkg/metrics/metric/metric.go

```diff
@@ -196,3 +241,84 @@ func (b Opts) GetConfigName() string {
 	}
 	return b.ConfigName
 }
+
+type Label struct {
+	Name string
+	// If defined, only these values are allowed.
+	Values Values
+}
+
+type Values map[string]struct{}
+
+func NewValues(vs ...string) Values {
+	vals := Values{}
+	for _, v := range vs {
+		vals[v] = struct{}{}
+	}
+	return vals
+}
+
+func NewLabel(name string, lvls ...string) Label {
```

Comment:
I'm not sure I love the function signature because at a the call site, it's not so obvious the first value is the label, and the remaining are the values. Could we remove `NewLabel` and use composite literals instead? The `NewValues` could still exist as a helper to construct it.

At the call site it would then look like:
```
metric.Labels{
  {
    Name: "outcome", 
    Values: metric.NewValues(LabelValueOutcomeSuccess, LabelValueOutcomeFail),
  },
}
```

It might even be nice to just use a `[]string` instead of `Values`, and handle the duplication check somewhere else, then the call site is very simple and easy to understand.

Reply:
Agreed, moved to using the composite literal approach, seems much cleaner now.